### PR TITLE
Fix for issue #411

### PR DESCRIPTION
**Pull request by birojnayak**
Fix for issue #411

### CODE REVIEW

**Comment by mconnew** [comment]
Can you add a test which uses this in a custom binding?

**Comment by birojnayak** [comment]
> Can you add a test which uses this in a custom binding? That includes a negative test to verify it fails when you use HttpsTransportBindingElement but only have an http endpoint.

@mconnew added the test(for +/- ve scenarios)

File: src/CoreWCF.Http/tests/BasicHttpTransportMessage.cs [hunk 1]
```diff
@@ -61,8 +61,40 @@ public void BasicHttpRequestReplyWithTransportMessageEchoString()
             }
         }
 
-       // [Fact, Description("transport-security-with-certificate-authentication")]
-       // TODO set up in container, tested locally and this works
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void BasicHttpsCustomBindingRequestReplyEchoString(bool httpsMode)
```
Comment (mconnew) [inline src/CoreWCF.Http/tests/BasicHttpTransportMessage.cs:67]: Can you rename parameter to useHttps as httpsMode is ambiguous.
